clean-ts/clean-ts.c

```diff
@@ -119,14 +119,18 @@ static int clean_ts(const char *infile, const char *outfile, int64_t npackets,
   output_streams = av_mallocz_array(input_stream_size, sizeof(*output_streams));
   size_t i;
   for (i = 0; i < input_stream_size; i++) {
-    output_streams[i] = avformat_new_stream(oc, input_streams[i]->codec->codec);
-    DPRINTF("%d: Copy from [0x%x]\n", output_streams[i]->index, input_streams[i]->id);
-    FAIL_IF_ERROR(avcodec_copy_context(output_streams[i]->codec, input_streams[i]->codec));
+    const AVCodec *codec = avcodec_find_encoder(input_streams[i]->codecpar->codec_id);
+    output_streams[i] = avformat_new_stream(oc, codec);
+    DPRINTF("%d: Copy from [0x%x]\n", output_streams[i]->index, input_streams[i]->index);
+    output_streams[i]->codecpar = input_streams[i]->codecpar;
     output_streams[i]->time_base = input_streams[i]->time_base;
   }
-  if (oc->oformat->flags & AVFMT_GLOBALHEADER) {
-    for (i = 0; i < input_stream_size; i++) {
-      output_streams[i]->codec->flags |= CODEC_FLAG_GLOBAL_HEADER;
+
+  for (i = 0; i < input_stream_size; i++) {
+    if (oc->oformat->flags & AVFMT_GLOBALHEADER) {
```

Comment:
Why did you move this branching into the loop? The condition `oc->oformat->flags & AVFMT_GLOBALHEADER` doesn't change during the loop.